plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/image/block.tsx

```diff
@@ -29,6 +29,27 @@ import './style.scss';
 import { BlockAttributes, ImageSizing, ProductImageContext } from './types';
 import { isTryingToDisplayLegacySaleBadge } from './utils';
 
+// PLACEHOLDER_IMG_SRC is in thumbnail size which may have aspect-ratio
+// set in Customizer and it may not reflect the aspect-ratio of the full size image.
+// So we're "creating" a full size placeholder URL from the thumbnail URL.
+// Example input: https://example.com/path/to/placeholder-150x150.png
+// Example output: https://example.com/path/to/placeholder.png
+const getPlaceholderSrc = ( showFullSize: boolean ) => {
+	return showFullSize
+		? PLACEHOLDER_IMG_SRC.replace( /-\d+x\d+(?=\.[^.]+$)/, '' )
+		: PLACEHOLDER_IMG_SRC;
+};
```

Comment:
If this is only required in this block, I guess it's fine doing it this way. But if we see this pattern being used in other places, I think we should register a new `PLACEHOLDER_IMG_SRC_FULL_SIZE` setting (or something along those lines) from the backend. `wc_placeholder_img_src()` allows passing the required size, so I think it would be more reliable to use the value generated from the backend than modifying the image path in the frontend.